ddtrace/tracer/option.go

```diff
@@ -38,6 +38,9 @@ var (
 	// defaultSocketDSD specifies the socket path to use for connecting to the statsd server.
 	// Replaced in tests
 	defaultSocketDSD = "/var/run/datadog/dsd.socket"
+
+	// defaultMaxTagsHeaderLen specifies the default maximum length of the X-Datadog-Tags header value.
+	defaultMaxTagsHeaderLen = 512
```

Comment:
How was this value chosen? Are we sure it's long enough? Are there limits to this for HTTP headers?

Let's add some more documentation here if we can.